brave/brave-core
# Copy text from image on Windows

Resolves https://github.com/brave/brave-browser/issues/29003

Window's OCR API for reading text from images.
Referred upstream's `shape_detection::TextDetectionImpl`.

To use OCR API, ocr engine is initialized with target language.
We show the result with user profile languge.
After that, we could try with another available language.

![image](https://user-images.githubusercontent.com/6786187/229010712-858ccb82-16dd-4e23-ab1b-f29178b0fc71.png)




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`TextRecognitionBrowserTest.TextRecognitionTest`

1. Search any image that contains text from search.brave.com
2. Choose Copy Text From Image item from context menu
3. Check dialog is shown with text
4. Check extracted text is copied to clipboard
